core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.errors
+
+import java.io.IOException
+
+import org.apache.spark.SparkException
+import org.apache.spark.shuffle.{FetchFailedException, ShuffleManager}
+import org.apache.spark.storage.{BlockId, BlockManagerId, BlockNotFoundException, UnrecognizedBlockId}
+
+object SparkCoreErrors {
+  def failedParseBlockIdError(name: String): Throwable = {
+    new UnrecognizedBlockId(name)
+  }
+
+  def taskHasNotLockedBlockError(currentTaskAttemptId: Long, blockId: BlockId): Throwable = {
+    new SparkException(s"Task $currentTaskAttemptId has not locked block $blockId for writing")
+  }
+
+  def blockDoesNotExistError(blockId: BlockId): Throwable = {
+    new SparkException(s"Block $blockId does not exist")
+  }
+
+  def cannotSaveBlockOnDecommissionExecutorError(blockId: BlockId): Throwable = {
+    new SparkException(s"Block $blockId cannot be saved on decommissioned executor")
+  }
+
+  def waitingForReplicationFinishError(t : scala.Throwable): Throwable = {
```

Comment:
`waitingForReplicationToFinishError(e: Throwable)`
nit: remove the white space after `t`